src/App.js

```diff
@@ -74,50 +74,92 @@ const App = () => {
+              setShutterPress={setShutterPress}
+            />
+            <HomePage
               setSearchFocus={setSearchFocus}
               searchFocus={searchFocus}
-              theme={theme}
             />
-
-            <BottomNav searchFocus={searchFocus} />
           </Route>
           <Route exact path="/category/:categoryId">
+            <TopNav
+              searchFocus={searchFocus}
+              theme={theme}
+              toggleTheme={toggleTheme}
+              shutterPress={shutterPress}
+              setShutterPress={setShutterPress}
+            />
             <CategoryPage categories={categories} materials={materials} />
-            <BottomNav searchFocus={searchFocus} />
           </Route>
           <Route exact path="/material/:materialId">
+            <TopNav
+              searchFocus={searchFocus}
+              theme={theme}
+              toggleTheme={toggleTheme}
+              shutterPress={shutterPress}
+              setShutterPress={setShutterPress}
+            />
             <MaterialPage />
-            <BottomNav searchFocus={searchFocus} />
           </Route>
           <Route exact path="/material/:materialId/locations">
+            <TopNav
+              searchFocus={searchFocus}
+              theme={theme}
+              toggleTheme={toggleTheme}
+              shutterPress={shutterPress}
+              setShutterPress={setShutterPress}
+            />
             <LocationsPage />
-            <BottomNav searchFocus={searchFocus} />
           </Route>
           <Route exact path="/subcategory/:subCategoryId">
+            <TopNav
+              searchFocus={searchFocus}
+              theme={theme}
+              toggleTheme={toggleTheme}
+              shutterPress={shutterPress}
+              setShutterPress={setShutterPress}
+            />
             <SubCategoryPage />
-            <BottomNav searchFocus={searchFocus} />
           </Route>
           <Route exact path="/intro">
             <TutorialPage />
           </Route>
           <Route exact path="/camera">
-            <CameraPage
-              setAppCluster={setAppCluster}
+            <TopNav
+              searchFocus={searchFocus}
+              theme={theme}
+              toggleTheme={toggleTheme}
               shutterPress={shutterPress}
-              setSearchFocus={setSearchFocus}
               setShutterPress={setShutterPress}
             />
-            <CameraNav
+
+            <CameraPage
+              theme={theme}
+              setAppCluster={setAppCluster}
               shutterPress={shutterPress}
               setShutterPress={setShutterPress}
               setSearchFocus={setSearchFocus}
+              setShutterPress={setShutterPress}
             />
           </Route>
           <Route exact path="/camera/results">
-            <ClusterPage appCluster={appCluster} />
-            <BottomNav searchFocus={searchFocus} />
+            <TopNav
```

Comment:
CodeClimate is yelling about this repetitive code, moving App.js maintainability from A to D. There isn't much that can be done here - this is an issue due to props drilling rather than grabbing from global store or context.